### ISSUE TITLE
ConcurrentModificationException in MainService.stopCommands()

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#316]
```
@MainService.java:749

Stacktrace:

java.lang.RuntimeException: Unable to stop service 
com.googlecode.gtalksms.MainService@406b3978: 
java.util.ConcurrentModificationException
at android.app.ActivityThread.handleStopService(ActivityThread.java:2086)
at android.app.ActivityThread.access$2900(ActivityThread.java:117)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1001)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:130)
at android.app.ActivityThread.main(ActivityThread.java:3687)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:507)
at 
com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:867)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:625)
at dalvik.system.NativeStart.main(Native Method)
Caused by: java.util.ConcurrentModificationException
at java.util.HashMap$HashIterator.nextEntry(HashMap.java:796)
at java.util.HashMap$KeyIterator.next(HashMap.java:823)
at com.googlecode.gtalksms.MainService.stopCommands(MainService.java:749)
at 
com.googlecode.gtalksms.MainService.teardownListenersForConnection(MainService.j
ava:818)
at com.googlecode.gtalksms.MainService.onDestroy(MainService.java:471)
at android.app.ActivityThread.handleStopService(ActivityThread.java:2069)
... 10 more

```

Original issue reported on code.google.com by `fschm...@gmail.com` on 15 Feb 2013 at 10:33

**Comment by GoogleCodeExporter** [comment]
Original comment by `Florent....@gmail.com` on 16 Aug 2014 at 6:04
- Changed state: **FixedInBeta**

**Comment by GoogleCodeExporter** [comment]
```
Released on Play Market as 5.0
```

Original comment by `Florent....@gmail.com` on 25 Oct 2014 at 6:23
- Changed state: **Deployed**
